### PR TITLE
Validate gossip message size

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Validate gossipsub max size of 10MB per spec

**Description**

- add Decompressor type to snappyjs
- call get length api and validate that in DataTransformSnappy

### CODE REVIEW

**Comment by dapplion** [comment]
We should upstream max size limits otherwise it's doing the work double. snappyjs already computes the uncompressed size

https://github.com/zhipeng-jia/snappyjs/blob/378fc9f8a119d53e4f0c25cd1d3295e28a3f1252/index.js#L69

```ts
  var decompressor = new SnappyDecompressor(compressed)
  var length = decompressor.readUncompressedLength()
  if (length === -1) {
    throw new Error('Invalid Snappy bitstream')
  }
```

This is how all other implementations work
- Lighthouse: https://github.com/sigp/lighthouse/blob/79db2d4deb6a47947699d8a4a39347c19ee6e5d6/beacon_node/lighthouse_network/src/types/pubsub.rs#L62
- Nimbus: https://github.com/status-im/nimbus-eth2/blob/stable/beacon_chain/networking/eth2_network.nim#L2219
- Teku: https://github.com/ConsenSys/teku/blob/b7fc946f9313801476b7fce2e4a02d91e7bb891b/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/encoding/SnappyPreparedGossipMessage.java#L122
- Prysm: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/p2p/message_id.go#L52

**Comment by twoeths** [comment]
I created https://github.com/zhipeng-jia/snappyjs/pull/14

File: packages/lodestar/src/network/gossip/encoding.ts [hunk 1]
```diff
@@ -74,6 +80,9 @@ export class DataTransformSnappy {
    * transformed data will then be used to create a `RawGossipsubMessage` to be sent to peers.
    */
   outboundTransform(topicStr: string, data: Uint8Array): Uint8Array {
+    if (data.length > this.maxSizePerMessage) {
+      throw Error(`ssz_snappy encoded data length ${length} > ${this.maxSizePerMessage}`);
+    }
```
Comment (dapplion) [inline packages/lodestar/src/network/gossip/encoding.ts:79]: Do we really need to limit outbound size? Do other implementations do it?

File: yarn.lock [hunk 2]
```diff
@@ -512,7 +512,78 @@
...
+
+"@chainsafe/lodestar-types@^0.37.0":
+  version "0.37.0"
+  resolved "https://registry.yarnpkg.com/@chainsafe/lodestar-types/-/lodestar-types-0.37.0.tgz#ebc585830051d30e0c877a50b431bfe02742db39"
+  integrity sha512-snDFbYUrtav9JawdMZk8ytr65Y8gZC4uSy60TCUCnHvvi+TqojZvjlX3eAwxktjlf8uailoZqDVNEh5ICzEfJw==
+  dependencies:
+    "@chainsafe/lodestar-params" "^0.37.0"
+    "@chainsafe/ssz" "^0.9.1"
+
+"@chainsafe/lodestar-utils@^0.37.0":
```
Comment (dapplion) [inline yarn.lock]: Please rebase to unstable and delete this diff

File: packages/lodestar/src/network/gossip/encoding.ts (hunk 1 above)
Comment (dapplion) [reply packages/lodestar/src/network/gossip/encoding.ts:79]: EDIT: We can keep it, since those messages would be rejected by our peers anyway

**Comment by dapplion** [comment]
Mark as ready if this is good to go!

**Review by dapplion — DISMISSED** [review]
LGTM!

**Review by dapplion — APPROVED** [review]

**Pull request merged**